library/network_connections.py

```diff
@@ -1133,6 +1135,28 @@ def connection_create(self, connections, idx, connection_current=None):
                 rr = NM.IPRoute.new(
                     r["family"], r["network"], r["prefix"], r["gateway"], r["metric"]
                 )
+                if isinstance(r["table"], int):
+                    NM.IPRoute.set_attribute(
+                        rr, "table", Util.GLib().Variant.new_uint32(r["table"])
+                    )
+                elif isinstance(r["table"], Util.STRING_TYPE):
+                    tableid = None
+                    with open("/etc/iproute2/rt_tables", "r") as f:
```

Comment:
the ArgValidator is supposed to validate and normalize the input configuration early on, before any changes are done. I think this parsing should happen there, it should not blindly pass on any strings. Yes, that would mean that ArgValidator needs a way to inject such "host configuration" (to access the name mapping). Add a singleton -- that can be stubbed for unit tests.

---

As I said on IRC, I find it a very bad idea to do this. It makes the configuration in the playbook dependent on what files are on the target host. Not to mention the added complexity for little benefit. Routing tables are numbers and not something to be hidden by the user.

Reply:
I updated the patch.

Reply:
Just for the record since we discussed this outside of the review: The table names exists because they make using tables more user friendly and I anticipate that users would like to use them. Also, the BZ for this feature mentioned named tables in their examples.